[Bug] [connector][hdfs] Caused by: java.io.EOFException: Unexpected end of input stream

### Search before asking

- [X] I had searched in the [issues](https://github.com/DTStack/chunjun/issues) and found no similar issues.


### What happened



读取hdfs报错：

```

2023-05-17 08:00:24,098 INFO  org.apache.flink.runtime.resourcemanager.slotmanager.DeclarativeSlotManager [] - Clearing resource requirements of job 85d8da75f4dd2ad5252234702905bc09
2023-05-17 08:00:24,096 INFO  org.apache.flink.runtime.executiongraph.ExecutionGraph       [] - Job Flink_Job (85d8da75f4dd2ad5252234702905bc09) switched from state FAILING to FAILED.
org.apache.flink.runtime.JobException: Recovery is suppressed by NoRestartBackoffTimeStrategy
	at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.handleFailure(ExecutionFailureHandler.java:139) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.getFailureHandlingResult(ExecutionFailureHandler.java:83) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.scheduler.DefaultScheduler.recordTaskFailure(DefaultScheduler.java:256) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.scheduler.DefaultScheduler.handleTaskFailure(DefaultScheduler.java:247) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.scheduler.DefaultScheduler.onTaskFailed(DefaultScheduler.java:240) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.scheduler.SchedulerBase.onTaskExecutionStateUpdate(SchedulerBase.java:738) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.scheduler.SchedulerBase.updateTaskExecutionState(SchedulerBase.java:715) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.scheduler.SchedulerNG.updateTaskExecutionState(SchedulerNG.java:78) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.runtime.jobmaster.JobMaster.updateTaskExecutionState(JobMaster.java:477) ~[flink-dist-1.16.1.jar:1.16.1]
	at sun.reflect.GeneratedMethodAccessor31.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_60]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_60]
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.lambda$handleRpcInvocation$1(AkkaRpcActor.java:309) ~[flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at org.apache.flink.runtime.concurrent.akka.ClassLoadingUtils.runWithContextClassLoader(ClassLoadingUtils.java:83) ~[flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcInvocation(AkkaRpcActor.java:307) ~[flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcMessage(AkkaRpcActor.java:222) ~[flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at org.apache.flink.runtime.rpc.akka.FencedAkkaRpcActor.handleRpcMessage(FencedAkkaRpcActor.java:84) ~[flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleMessage(AkkaRpcActor.java:168) ~[flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:24) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:20) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at scala.PartialFunction.applyOrElse(PartialFunction.scala:123) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at scala.PartialFunction.applyOrElse$(PartialFunction.scala:122) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.japi.pf.UnitCaseStatement.applyOrElse(CaseStatements.scala:20) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:171) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.actor.Actor.aroundReceive(Actor.scala:537) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.actor.Actor.aroundReceive$(Actor.scala:535) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:220) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.actor.ActorCell.invoke(ActorCell.scala:548) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.dispatch.Mailbox.run(Mailbox.scala:231) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243) [flink-rpc-akka_2f94a0e2-3491-4ab4-b7a8-5f3a6f000c01.jar:1.16.1]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_60]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_60]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_60]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [?:1.8.0_60]
Caused by: java.io.EOFException: Unexpected end of input stream
	at org.apache.hadoop.io.compress.DecompressorStream.decompress(DecompressorStream.java:165) ~[hadoop-common-3.2.1U20.jar:?]
	at org.apache.hadoop.io.compress.DecompressorStream.read(DecompressorStream.java:105) ~[hadoop-common-3.2.1U20.jar:?]
	at java.io.InputStream.read(InputStream.java:101) ~[?:1.8.0_60]
	at org.apache.hadoop.util.LineReader.fillBuffer(LineReader.java:182) ~[hadoop-common-3.2.1U20.jar:?]
	at org.apache.hadoop.util.LineReader.readDefaultLine(LineReader.java:218) ~[hadoop-common-3.2.1U20.jar:?]
	at org.apache.hadoop.util.LineReader.readLine(LineReader.java:176) ~[hadoop-common-3.2.1U20.jar:?]
	at org.apache.hadoop.mapred.LineRecordReader.skipUtfByteOrderMark(LineRecordReader.java:215) ~[hadoop-mapreduce-client-core-3.2.1U20.jar:?]
	at org.apache.hadoop.mapred.LineRecordReader.next(LineRecordReader.java:253) ~[hadoop-mapreduce-client-core-3.2.1U20.jar:?]
	at org.apache.hadoop.mapred.LineRecordReader.next(LineRecordReader.java:48) ~[hadoop-mapreduce-client-core-3.2.1U20.jar:?]
	at com.dtstack.chunjun.connector.hdfs.source.BaseHdfsInputFormat.reachedEnd(BaseHdfsInputFormat.java:109) ~[chunjun-connector-hdfs-master.jar:?]
	at com.dtstack.chunjun.source.DtInputFormatSourceFunction.run(DtInputFormatSourceFunction.java:123) ~[chunjun-core.jar:?]
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:67) ~[flink-dist-1.16.1.jar:1.16.1]
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:333) ~[flink-dist-1.16.1.jar:1.16.1]
2023-05-17 08:00:24,123 INFO  org.apache.flink.runtime.checkpoint.CheckpointCoordinator    [] - Stopping checkpoint coordinator for job 85d8da75f4dd2ad5252234702905bc09.
2023-05-17 08:00:24,156 INFO  org.apache.flink.runtime.dispatcher.MiniDispatcher           [] - Job 85d8da75f4dd2ad5252234702905bc09 reached terminal state FAILED.
org.apache.flink.runtime.JobException: Recovery is suppressed by NoRestartBackoffTimeStrategy
	at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.handleFailure(ExecutionFailureHandler.java:139)
	at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.getFailureHandlingResult(ExecutionFailureHandler.java:83)
	at org.apache.flink.runtime.scheduler.DefaultScheduler.recordTaskFailure(DefaultScheduler.java:256)
	at org.apache.flink.runtime.scheduler.DefaultScheduler.handleTaskFailure(DefaultScheduler.java:247)
	at org.apache.flink.runtime.scheduler.DefaultScheduler.onTaskFailed(DefaultScheduler.java:240)
	at org.apache.flink.runtime.scheduler.SchedulerBase.onTaskExecutionStateUpdate(SchedulerBase.java:738)
	at org.apache.flink.runtime.scheduler.SchedulerBase.updateTaskExecutionState(SchedulerBase.java:715)
	at org.apache.flink.runtime.scheduler.SchedulerNG.updateTaskExecutionState(SchedulerNG.java:78)
	at org.apache.flink.runtime.jobmaster.JobMaster.updateTaskExecutionState(JobMaster.java:477)
	at sun.reflect.GeneratedMethodAccessor31.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.lambda$handleRpcInvocation$1(AkkaRpcActor.java:309)
	at org.apache.flink.runtime.concurrent.akka.ClassLoadingUtils.runWithContextClassLoader(ClassLoadingUtils.java:83)
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcInvocation(AkkaRpcActor.java:307)
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcMessage(AkkaRpcActor.java:222)
	at org.apache.flink.runtime.rpc.akka.FencedAkkaRpcActor.handleRpcMessage(FencedAkkaRpcActor.java:84)
	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleMessage(AkkaRpcActor.java:168)
	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:24)
	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:20)
	at scala.PartialFunction.applyOrElse(PartialFunction.scala:123)
	at scala.PartialFunction.applyOrElse$(PartialFunction.scala:122)
	at akka.japi.pf.UnitCaseStatement.applyOrElse(CaseStatements.scala:20)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:171)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:220)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
	at akka.actor.ActorCell.invoke(ActorCell.scala:548)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.io.EOFException: Unexpected end of input stream
	at org.apache.hadoop.io.compress.DecompressorStream.decompress(DecompressorStream.java:165)
	at org.apache.hadoop.io.compress.DecompressorStream.read(DecompressorStream.java:105)
	at java.io.InputStream.read(InputStream.java:101)
	at org.apache.hadoop.util.LineReader.fillBuffer(LineReader.java:182)
	at org.apache.hadoop.util.LineReader.readDefaultLine(LineReader.java:218)
	at org.apache.hadoop.util.LineReader.readLine(LineReader.java:176)
	at org.apache.hadoop.mapred.LineRecordReader.skipUtfByteOrderMark(LineRecordReader.java:215)
	at org.apache.hadoop.mapred.LineRecordReader.next(LineRecordReader.java:253)
	at org.apache.hadoop.mapred.LineRecordReader.next(LineRecordReader.java:48)
	at com.dtstack.chunjun.connector.hdfs.source.BaseHdfsInputFormat.reachedEnd(BaseHdfsInputFormat.java:109)
	at com.dtstack.chunjun.source.DtInputFormatSourceFunction.run(DtInputFormatSourceFunction.java:123)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:67)
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:333)
2023-05-17 08:00:24,167 INFO  org.apache.flink.runtime.dispatcher.MiniDispatcher           [] - Job 85d8da75f4dd2ad5252234702905bc09 has been registered for cleanup in the JobResultStore after reaching a terminal state.
2023-05-17 08:00:24,182 INFO  org.apache.flink.runtime.jobmaster.JobMaster                 [] - Stopping the JobMaster for job 'Flink_Job' (85d8da75f4dd2ad5252234702905bc09).

```



初步分析：读取的hdfs路径下有空文件。

### What you expected to happen

--

### How to reproduce




hdfs 路径下文件列表：

```

-rw-r-----   3 test test      34879 2023-05-16 05:11 hdfs://namenode.test.net:9000/home/test/client_encrypt_common_statistics/2023-05-16/0500/uninst-.361_2023-05-16-05-10.gz
-rw-r-----   3 test test          0 2023-05-16 06:51 hdfs://namenode.test.net:9000/home/test/client_encrypt_common_statistics/2023-05-16/0500/uninst-ltrans1-2023-05-16-05-50-empty.gz

```



运行的sql"

```

CREATE TABLE source
(
  `raw_message` string
) WITH (
      'file-type' = 'text',
      'filter-regex' = '',
      'field-delimiter' = '',
      'default-fs' = 'hdfs://namenode.test.net:9000',
      'encoding' = 'utf-8',
      'properties.fs.defaultFS' = 'hdfs://namenode.test.net:9000',
      'properties.dfs.client.failover.proxy.provider.ns' = 'org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider',
      'path' = 'hdfs://namenode.test.net:9000/home/test/client_encrypt_common_statistics/2023-05-16/0500/',
      'connector' = 'hdfs-x',
      'properties.hadoop.user.name' = 'test',
      'properties.fs.hdfs.impl' = 'org.apache.hadoop.fs.viewfs.ViewFileSystem',
      'properties.fs.hdfs.impl.disable.cache' = 'true'
      );
CREATE TABLE sink
(
  `data` string
) WITH (
      'connector' = 'stream-x',
      'print' = 'true'
      );
;
insert into sink select  * from source;


```

### Anything else

_No response_

### Version

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/chunjun-connectors/chunjun-connector-hdfs/src/main/java/com/dtstack/chunjun/connector/hdfs/source/BaseHdfsInputFormat.java
+++ b/chunjun-connectors/chunjun-connector-hdfs/src/main/java/com/dtstack/chunjun/connector/hdfs/source/BaseHdfsInputFormat.java
@@ -60,6 +60,7 @@ public abstract class BaseHdfsInputFormat extends BaseRichInputFormat {
     protected transient org.apache.hadoop.mapred.InputFormat inputFormat;
     protected transient JobConf hadoopJobConf;
     protected transient RecordReader recordReader;
+    protected String currentReadFilePath;
 
     @Override
     public InputSplit[] createInputSplitsInternal(int minNumSplits) throws IOException {
@@ -106,7 +107,14 @@ public abstract class BaseHdfsInputFormat extends BaseRichInputFormat {
     @Override
     @SuppressWarnings("unchecked")
     public boolean reachedEnd() throws IOException {
-        return !recordReader.next(key, value);
+        boolean reachedEnd;
+        try {
+            reachedEnd = !recordReader.next(key, value);
+        } catch (IOException e) {
+            throw new ChunJunRuntimeException(
+                    e.getMessage() + ", current read file path: " + currentReadFilePath);
+        }
+        return reachedEnd;
     }
 
     @Override

--- a/chunjun-connectors/chunjun-connector-hdfs/src/main/java/com/dtstack/chunjun/connector/hdfs/source/HdfsTextInputFormat.java
+++ b/chunjun-connectors/chunjun-connector-hdfs/src/main/java/com/dtstack/chunjun/connector/hdfs/source/HdfsTextInputFormat.java
@@ -40,6 +40,7 @@ import org.apache.hadoop.mapred.TextInputFormat;
 
 import java.io.IOException;
 import java.security.PrivilegedAction;
+import java.util.ArrayList;
 import java.util.List;
 
 @Slf4j
@@ -63,11 +64,15 @@ public class HdfsTextInputFormat extends BaseHdfsInputFormat {
                 inputFormat.getSplits(hadoopJobConf, minNumSplits);
 
         if (splits != null) {
-            HdfsTextInputSplit[] hdfsTextInputSplits = new HdfsTextInputSplit[splits.length];
+            List<HdfsTextInputSplit> splitList = new ArrayList<>();
             for (int i = 0; i < splits.length; ++i) {
-                hdfsTextInputSplits[i] = new HdfsTextInputSplit(splits[i], i);
+                HdfsTextInputSplit split = new HdfsTextInputSplit(splits[i], i);
+                if (split.getTextSplit().getLength() == 0) {
+                    continue;
+                }
+                splitList.add(split);
             }
-            return hdfsTextInputSplits;
+            return splitList.toArray(new HdfsTextInputSplit[splitList.size()]);
         }
         return null;
     }
@@ -104,6 +109,7 @@ public class HdfsTextInputFormat extends BaseHdfsInputFormat {
                 super.inputFormat.getRecordReader(fileSplit, super.hadoopJobConf, Reporter.NULL);
         super.key = new LongWritable();
         super.value = new Text();
+        super.currentReadFilePath = ((FileSplit) fileSplit).getPath().toString();
     }
 
     @Override

